File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void) 
{
    float change;
    do 
    {
        change = get_float("Change owed: ");
    } 
    while (change < 0);
    
    // convert change to int by multiplying by 100
    int c = round(change * 100);
    
    // initialize n to 0, remaining to c
    int n = 0;
    int remaining = c;
    
    // check how many 25c coins are needed
    n += remaining / 25;
    remaining -= (remaining / 25) * 25;
    
    // add how many 10c coins are needed
    n += remaining / 10;
    remaining -= (remaining / 10) * 10;
    
    // add how many 5c coins are needed
    n += remaining / 5;
    remaining -= (remaining / 5) * 5;
    
    // add how many 1c coins are needed
    n += remaining;
    
    printf("%i\n", n);
}

File: pset6/cash/cash.py
from cs50 import get_float

# get user input 
while True:
    amount = get_float("Change owed: ")
    
    # keep prompting unless the value is > 0
    if amount > 0:
        break
    
# convert amount to cents
cents = int(amount * 100)

coins = 0
for denom in [25, 10, 5]:
    # number of coins for current denoms
    denom_coins = cents // denom
    coins += denom_coins
    
    # remaining amount
    cents -= denom_coins * denom

# add remaining cents for 1c coins
coins += cents

print(coins)